### PR TITLE
Testing improvements for climate/food logic

### PR DESCRIPTION
**Pull request by benmcroberts**
* Add extra tests to check climate/food logic.
* Clean-up 'board_test'
* Fix issue with negative food in watering hole

### CODE REVIEW

File: core/tests/phase_select_food_and_climate_test.py [hunk 1]
```diff
@@ -9,6 +9,57 @@ def setUp(self):
         self.game = Game(["Smith", "Gould"])
         self.phase = SelectFoodAndClimatePhase(self.game)
```
Comment (a-toms) [inline core/tests/phase_select_food_and_climate_test.py]: Suggest changing self.phase to self.climate_phase. May be easier to understand on first glance.

File: core/tests/phase_select_food_and_climate_test.py [hunk 2]
```diff
@@ -9,6 +9,57 @@ def setUp(self):
...
+
+        self.assertEqual(expected_climate_name, actual_climate.name)
+
+    def __test_food_effects(self, net_food):
+        expected_watering_hole = max(net_food, 0)
+
+        self.assertEqual(expected_watering_hole, self.game.board.watering_hole)
+
+    def test_food_and_climate_effects(self):
+        for food in [-5, -1, 0, 1, 10]:
```
Comment (a-toms) [inline core/tests/phase_select_food_and_climate_test.py]: Suggest slight update.

**Review by a-toms — COMMENTED** [review]
Read.

File: core/tests/phase_select_food_and_climate_test.py (hunk 1 above)
Comment (benmcroberts) [reply core/tests/phase_select_food_and_climate_test.py]: Agree

**Review by a-toms — APPROVED** [review]

**Pull request merged**